### DISCUSSION TITLE
Mocking dependencies using vitest

### DISCUSSION THREAD

**Discussion by carlosmoliv** [vitest-dev/vitest#3747] [Q&A]
Hello everyone, how's it going?

The following mock works when I use Jest, but it doesn't work with Vitest. I understand that Vitest operates differently in this scenario, but I haven't been able to find a solution yet.

![image](https://github.com/vitest-dev/vitest/assets/52833750/80a63bfb-cc08-41f3-b4a8-a8c783839ff9)

**Comment by samhirtarif** [comment]
Is `AwsS3Adapter` a function?

**Reply by carlosmoliv** [reply]
Hey, it's a class

**Comment by jsmrcaga** [comment]
@carlosmoliv did you ever find a solution? I'm facing a very similar issue trying to mock an entire module dependency

**Reply by carlosmoliv** [reply]
It's a project that I don't work on anymore, but I checked the code to refresh my memory. I used dynamic importing:

 ```ts
it('should returns 200 and the document added to the program', async () => {
    // Arrange
    const program = await programDSL.createProgram(
      makeFakeCreateProgramRequest(),
    );

    const awsS3Adapter = await import(
      '@infra/gateways/storage/aws-s3.adapter'
    );

    awsS3Adapter.AwsS3Adapter.prototype.uploadFile = vi
      .fn()
      .mockResolvedValue('any_file_path');

    // Rest of the code...
});
```

At the top of the test suite I called vi.mock 

```ts
vi.mock('@infra/storage/s3-adapter/aws-s3.adapter');

describe('Program Routes', () => {

// Rest of the code
```
